### PR TITLE
Fix test cases bug

### PR DESCRIPTION
**Pull request by minghowC**
Please help fixing bugs related to test cases: 
Screenshot 2024-03-26 at 13 33 44

### CODE REVIEW

**Comment by eunrcn** [comment]
minghow, i see what you have done here, thank you for changing the tags to enum and lets fix the testcases together on thursday. you are so kind-hearted and free-spirited, i look forward to your skit next tuesday and i hope to buy more of your morning coffees in the future. thank you for coming to class so early and paying attention to class when i dont. i truly value your presence even when you are silent and just listening so don't you dare think i dont notice you. lets work hard together for the remaining weeks and afterwards you will never have someone scream MINGHOWWWW at you ever again hAHAAHHAA

File: src/main/java/seedu/address/model/person/Tag.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+enum Tags {
+    NR("No Reply"),
+    OA("Online Assessment"),
+    I("Interview"),
+    R("Reject"),
+    O("Offer");
```
Comment (ashleygoh1) [inline src/main/java/seedu/address/model/person/Tag.java:11]: great use of enum
look better than using if else

File: src/main/java/seedu/address/model/person/Tag.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    /*
+     * Function overloading - isValidTag(String) & isValidTag(Tag)
+     */
+    public static boolean isValidTag(Tag tag) {
+        return Tags.isValidTag(tag.getTagName());
+    }
+
+    public String getTagName() {
+        return this.value.descriptiveName;
+    }
```
Comment (ashleygoh1) [inline src/main/java/seedu/address/model/person/Tag.java:103]: good use of method to ensure descriptive is private and unable to be access or changed directly by other files

File: src/main/java/seedu/address/model/person/Tag.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+enum Tags {
+    NR("No Reply"),
+    OA("Online Assessment"),
+    I("Interview"),
+    R("Reject"),
+    O("Offer");
+
+
+    public final String descriptiveName;
```
Comment (ashleygoh1) [inline src/main/java/seedu/address/model/person/Tag.java:14]: maybe can make this private, so it wont be modified or accessed by other files

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 4]
```diff
@@ -65,7 +65,7 @@ public JsonAdaptedPerson(Person source) {
         phone = source.getPhone().value;
         email = source.getEmail().value;
         address = source.getAddress().value;
-        tag = source.getTag().tagName;
+        tag = source.getTag().getTagName();
```
Comment (ashleygoh1) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java:68]: good use of calling method instead of modifying the variable tagName directly

**Review by ashleygoh1 — COMMENTED** [review]
LGTM!! great job in implementing the tag enum! The code was clean and easy to follow.

**Pull request merged**